task/gradle/0.3/gradle.yaml

```diff
@@ -0,0 +1,71 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: gradle
+  labels:
+    app.kubernetes.io/version: "0.3"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/displayName: Gradle
+    tekton.dev/categories: Build Tools
+    tekton.dev/tags: build-tool
+    tekton.dev/platforms: "linux/amd64,linux/s390x,linux/ppc64le"
+spec:
+  description: >-
+    This Task can be used to run a Gradle build.
+
+  workspaces:
+    - name: source
+      description: The workspace consisting of the gradle project.
+    - name: gradle-config
+      description: The workspace consisting of the custom gradle properties provided by the user.
+      optional: true
+  params:
+    - name: GRADLE_IMAGE
+      description: Gradle base image.
+      type: string
+      default: docker.io/library/gradle:7.6-jdk8@sha256:ebda657f66e04cfa9f152896b4fa70bea4b80561a836adddca3ed85d622f8898
+    - name: GRADLE_CONFIG_IMAGE
+      description: Gradle configuration image.
+      type: string
+      default: registry.access.redhat.com/ubi9-minimal:9.1
+    - name: PROJECT_DIR
+      description: The directory containing build.gradle
+      type: string
+      default: "."
+    - name: TASKS
+      description: 'The gradle tasks to run (default: build)'
+      type: array
+      default:
+        - build
+  steps:
+    - name: gradle-config
+      image: $(params.GRADLE_CONFIG_IMAGE)
+      script: |
+        #!/usr/bin/env bash
+        if [[ "$(workspaces.gradle-config.bound)" == "true" ]]; then
+          echo "gradle-config workspace is bound"
+          if [[ -f $(workspaces.gradle-config.path)/gradle.properties ]]; then
+            echo "copying gradle.properties to /home/gradle"
+            cp "$(workspaces.gradle-config.path)/gradle.properties" /home/gradle
+          fi
+        fi
+        whoami
```

Comment:
do we need this `whoami` ?

Reply:
No, it shouldn't be there. IAt first I ran the Task as user 1000 but that doesn't work in your test environment.
I was curious as which user the task ran.